Plugin.cs

```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
             if (!this.configuration.alwaysOn && (!this.configuration.smart || !Retainer())) return;
             var listing = MarketBoardCurrentOfferings.Read(dataPtr);
             var i = 0;
-            if (this.configuration.hq && this.items.Single(j => j.RowId == listing.ItemListings[0].CatalogId).CanBeHq)
-            {
+
+            // Load addon ptr from memory
+            var toolTipPtr = pi.Framework.Gui.GetUiObjectByName("ItemDetail", 1) + 0x258;
+            var toolTipItemName = GetSeStringText(GetSeString(toolTipPtr));
+            bool isCurrentItemHQ = false;
+
+            // Checks for HQ symbol in item name
+            if (toolTipItemName.Substring(toolTipItemName.Length - 1) == "")
+                isCurrentItemHQ = true;
+            else
+                isCurrentItemHQ = false;
+
+            if (isCurrentItemHQ) {
                 while (i < listing.ItemListings.Count && !listing.ItemListings[i].IsHq) i++;
                 if (i == listing.ItemListings.Count) return;
+            } else {
+                while (i < listing.ItemListings.Count && listing.ItemListings[i].IsHq) i++;
+                if (i == listing.ItemListings.Count) return;
```

Comment:
What's going on here? If `isCurrentItemHQ == false` we should just proceed with `i = 0`, right? I don't see why your code needs to change any of the logic after deciding whether to get lowest vs lowest HQ.

Reply:
If it is false - meaning the current item we're looking it is not HQ
As far as I understand, the i refers to the index of the listing

The only thing I changed this part was the !(boolean) before the listing.ItemListings[i].IsHq depends on the quality
I left the logic the same

Reply:
I think this is silly. The reason why we have the loop in the HQ case is so we skip over the NQ items when figuring out the cheapest listing to undercut.
Putting this loop in the NQ case means we skip over the HQ items when deciding when to undercut, so if an item has a 1000g HQ listing and then a 2000g NQ listing, you would undercut the 2000g one. I think most people would want to undercut the cheapest listing regardless of its NQ or HQ if the item they have is NQ, so we don't want the loop in the first place (i.e. if we have an NQ item, we always undercut the cheapest item (i = 0), not the cheapest NQ item (i = whatever you end up from this loop))